code/datums/gamemode/role/ninja.dm

```diff
@@ -474,4 +504,44 @@
 		if("animeposter5")
 			name = "EVA poster"
 		if("animeposter6")
-			name = "Mob Psycho poster"
\ No newline at end of file
+			name = "Mob Psycho poster"
+
+//Special Katana. Main katana in weaponry.dm
+/obj/item/weapon/katana/hesfast
+	siemens_coefficient = 0
+	cant_drop = TRUE
+	actions_types = list(/datum/action/item_action/toggle_teleport)
+
+/obj/item/weapon/katana/hesfast/New()
+	..()
+	daemon = new /datum/daemon/teleport(src,"sound/weapons/shing.ogg", "Pshh... nothing personnel... kid...")
+
+/obj/item/weapon/katana/hesfast/examine(mob/user)
+	..()
+	if(!isninja(user))
+		return
+	if(!daemon)
+		return
+	to_chat(user, "<span class='notice'>This katana has an ancient power dwelling inside of it!</span>")
+	var/message = "<span class='notice'>"
+	if(daemon.cooldown < world.time)
+		message += "Oh yeah, the ancient power stirs. This is the katana that will pierce the heavens!"
+	else
+		var/cooldowncalculated = round((daemon.cooldown - world.time)/10)
+		message += "Your steel has unleashed its dark and unwholesome power, so it's tapped out right now. It'll be ready again in [cooldowncalculated] seconds."
+	if(daemon.active)
+		message += " Your teleport is active, just like Goku's Shunkan Idou (Instant Transmission for Gaijin).</span>"
+	else
+		message += " Your teleport is inactive, just like a no-warp trap room in Aincrad.</span>"
+	to_chat(user, "[message]")
+
+/obj/item/weapon/katana/hesfast/suicide_act(mob/user)
+	to_chat(viewers(user), "<span class='danger'>[user] is slicing \his chest open with the [src.name]! It looks like \he's trying to commit sudoku.</span>")
```

Comment:
This should probably be `visible_message()` but most of the other `suicide_act()`s do it this way so it's OK